src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -798,201 +915,183 @@ def get_traces(self, start_frame, end_frame, channel_indices):
+        Additional parameters for the storage backend (e.g. AWS credentials) used for "zarr" stream_mode.
+    use_pynwb: bool, default: False
+        Uses the pynwb library to read the NWB file. Setting this to False, the default, uses h5py
+        to read the file. Using h5py can improve performance by bypassing some of the PyNWB validations.
 
-            # Get sampling frequency
-            if "starting_time" in electrical_series.keys():
-                t_start = electrical_series["starting_time"][()]
-                sampling_frequency = electrical_series["starting_time"].attrs["rate"]
-            elif "timestamps" in electrical_series.keys():
-                timestamps = electrical_series["timestamps"][:]
-                t_start = timestamps[0]
-                sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
+    Returns
+    -------
+    sorting: NwbSortingExtractor
+        The sorting extractor for the NWB file.
+    """
 
-            self.t_start = t_start
+    extractor_name = "NwbSorting"
+    mode = "file"
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+    name = "nwb"
 
-        assert (
-            sampling_frequency is not None
-        ), "Couldn't load sampling frequency. Please provide it with the 'sampling_frequency' argument"
-        assert (
-            self.t_start is not None
-        ), "Couldn't load a starting time for the sorting. Please provide it with the 't_start' argument"
+    def __init__(
+        self,
+        file_path: str | Path,
+        electrical_series_path: str | None = None,
+        sampling_frequency: float | None = None,
+        samples_for_rate_estimation: int = 1_000,
+        unit_table_path: str = "units",
+        stream_mode: str | None = None,
+        stream_cache_path: str | Path | None = None,
+        *,
+        t_start: float | None = None,
+        cache: bool = False,
+        storage_options: dict | None = None,
+        use_pynwb: bool = False,
+    ):
+        self.stream_mode = stream_mode
+        self.stream_cache_path = stream_cache_path
+        self.electrical_series_path = electrical_series_path
+        self.file_path = file_path
+        self.t_start = t_start
+        self.internal_sampling_frequency = sampling_frequency
+        self.storage_options = storage_options
+        self.units_table = None
 
-        units_table = hdf5_file["units"]
-        spike_times_data = units_table["spike_times"]
-        spike_times_index_data = units_table["spike_times_index"]
+        if self.stream_mode is None:
+            self.backend = get_backend_from_local_file(file_path)
+        else:
+            if self.stream_mode == "zarr":
+                self.backend = "zarr"
+            else:
+                self.backend = "hdf5"
 
-        if "unit_name" in units_table:
-            unit_ids = units_table["unit_name"]
+        if use_pynwb:
```

Comment:
Like Sam I think that the normal import errors are enough and we don't require special messages for these ones.